adserver/decisionengine/backends.py

```diff
@@ -393,13 +393,35 @@ def select_flight(self):
 
         return None
 
+    def get_ad_ctr_weight(self, ad):
+        """Apply the ad weighting factor based on the sampled CTR."""
+        weights = {
+            0.060: 1,
+            0.070: 2,
+            0.080: 3,
+            0.090: 4,
+            0.100: 5,
+            0.125: 6,
+            0.150: 7,
+            0.175: 8,
+            0.200: 9,
+            0.250: 10,
+        }
+        ad_weighting = 0
+        for threshold, weight in weights.items():
+            if ad.sampled_ctr >= threshold and weight > ad_weighting:
+                ad_weighting = weight
+
+        return ad_weighting
+
     def select_ad_for_flight(self, flight):
         """
         Choose an ad from the selected flight.
```

Comment:
```suggestion
        Choose an ad from the selected flight filtered requested ``self.ad_types``. 
```

I wanted to make sure this is downstream of ad_type,
since I saw a campaign with text ads doing ~4x better CTRs.
This implementation looks great though 💯